src/user/service.ts

```diff
@@ -274,3 +273,12 @@ function isExistingUser(user: User | NewUser): user is User {
   return (<User> user).id !== undefined;
 
 }
+
+function getByHref(href: string): string {
+
+  const fullUrl = new URL(href);
+  if (fullUrl) {
+    return fullUrl.pathname;
+  }
+  return href;
+}
```

Comment:
Looking better!

The name is still a little weird though... `getByHref` .... get 'what' by href?  Since it's a users service, the name suggest to me it's basically the same as findByHref... try to think about what this actually does.

Second, could you write a unit test for this? We don't have a lot of those in a12nserver, but it would be nice to see some proof that this function fully works.